rfc/0009-rancher-integration-charts.md

```diff
@@ -0,0 +1,174 @@
+
+Modify our charts to meet those. This includes (but is not exhaustive to, as it
+is a moving target):
+
+- Set minimum required Rancher chart annotations:
+  * `catalog.cattle.io/namespace`: should be postfixed with `-system` for UI
+    filtering. When creating v2 of the chart, the namespace should be different
+    than on v1, to handle upgrades/cleanup independently.
+  * `catalog.cattle.io/release-name`: If unset, release name can be changed in UI.
+  * `catalog.cattle.io/certified`
+  * `catalog.cattle.io/certified`
+  * `catalog.cattle.io/experimental`
+  * `catalog.cattle.io/ui-component`
+  * `catalog.cattle.io/display-name`
+  * `catalog.cattle.io/os`
+- Set `catalog.cattle.io/rancher-version` annotation as is required for
+  user-facing charts.
+- Provide `chart/appreadme.md`.
+- Provide logo for the chart by settings `icon` in `Chart.yaml`.
+- Rancher automatically creates a list of container images from charts for
+  airgap. Adopt Rancher's `repository` and `tag` values in values.yml.
+- Provide our own `serviceaccount`.
+- Define k8s min/max version with `kubeVersion` in `Chart.yaml`.
+- Add scheduling options at a pod (affinity) and node level
+  (taints/tolerations/selectors).
+- Provide `annotations.catalog.cattle.io/upstream-version`. This could be
+  automatically bumped on release.
+- Provide `questions.yaml` for defaults on Rancher Explorer UI.
+
+The Kubewarden stack has the following dependencies:
+- Designated main chart: `kubewarden-controller`. This will be the main charts
+  to build a Rancher package from.
+- Hard dependencies: 
+  * CRDs: in `kubewarden-crds` chart. On the main Kubewarden chart
+    (`kubewarden-controller`), set `catalog.cattle.io/auto-install:
+    kubewarden-crds=match` that points to the right CRD version.
+    This annotation accepts `match`, or a hardcoded version (e.g: `1.1.0`)
+    We ought to set it to a hardcoded version, to be able to release patch
+    versions.  of `kubewarden-controller` independently.
+  * `cert-manager`: considered a precondition of Rancher Explorer and out of
+    its scope.
+- Soft (optional) dependencies:
+  `open-telemetry/opentelemetry-operator`
+  `grafana`
+  `kube-prometheus-stack`
+  `jaeger-operator`
+  `kubewarden-defaults`.
+
+  For the ones depending on CRDs (`kubewarden-defaults`,
+    `opentelemetry-operator`, `jaeger-operator`), Rancher charts follow a 3 step
+  process:
+  * Wrap the custom resource in a conditional that is tied to a values.yaml
+    parameter.
+  * Talk to the Rancher Explorer UI team to implement a solution and create a
+  dashboard issue. Make them aware that if a certain CRD is defined we want to
+  flip the new added value to true. If there is configuration available on that
+  resource then it would be grayed out in the UI if the CRD is not available,
+  and editable if the CRD is available.
```

Comment:
I don't understand if Rancher Manager is already able to handle charts that depend on CRDs or if that's something new they have to implement

Reply:
They handle hard dependencies via chart annotations, and soft dependencies via logic in the UI, depending on which values one sets.
Tried to explain it better in https://github.com/kubewarden/rfc/pull/11/commits/914babd85855ed860df743e6091cdc093e244493.